src/shogun/machine/IterativeMachine.h

```diff
@@ -0,0 +1,127 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Shubham Shukla
+ */
+
+#ifndef _ITERATIVEMACHINE_H__
+#define _ITERATIVEMACHINE_H__
+
+#include <shogun/lib/config.h>
+
+#include <shogun/base/progress.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/lib/common.h>
+#include <shogun/machine/LinearMachine.h>
+
+namespace shogun
+{
+	class CFeatures;
+	class CLabels;
+
+	/** @brief Class IterativeLinearMachine implements an iterative model
```

Comment:
You can remove the class name
"Mix-in class that implements an iterative model whose ..."